understanding-asynchronous-programming/example_3.py

```diff
@@ -27,18 +29,16 @@ def main():
     tasks = [task("One", work_queue), task("Two", work_queue)]
 
     # Run the tasks
-    et = ET()
     done = False
-    while not done:
-        for t in tasks:
-            try:
-                next(t)
-            except StopIteration:
-                tasks.remove(t)
-            if len(tasks) == 0:
-                done = True
-
-    print(f"\nTotal elapsed time: {et():.1f}")
+    with Timer(text='\nTotal elapsed time: {:.2f}'):
+        while not done:
+            for t in tasks:
+                try:
+                    next(t)
+                except StopIteration:
+                    tasks.remove(t)
+                if len(tasks) == 0:
```

Comment:
I see this comes from the original code, so I'll let you be the judge if it's worth changing:

I believe it's more "pythonic" to test directly on the falsyness of the list:

```
if not tasks:
    done = True
```

Reply:
An even bigger change (which would probably force you to update some of the explaining text as well) could be to refactor out the `done` variable completely:

```
with Timer(text="\nTotal elapsed time: {:.1f}"):
    while True:
        for t in tasks:
            try:
                next(t)
            except StopIteration:
                tasks.remote(t)
            if not tasks:
                break
```

Personally, I find this cleaner than using the extra variable, but I'm not sure it's as obvious for everybody? Again, I'll leave it for you to decide.

Reply:
I'm going to leave this as it is, though I agree with your choices. Since the article is already out there this way, I'm going to leave it be for the moment. Next time... :)